frontend/src/exporter/exporterViewLogic.ts

```diff
@@ -1,17 +1,99 @@
 // E.g. api.ts uses this to add the sharing_access_token
 export const exporterViewLogic = kea<exporterViewLogicType>([
-    path(() => ['scenes', 'exporter', 'exporterLogic']),
+    path(['exporter', 'exporterViewLogic']),
     props({} as ExportedData),
-    selectors(() => ({
-        exportedData: [() => [(_, props) => props], (props): ExportedData => props],
+    connect((props: ExportedData) => ({
+        actions: [
+            heatmapDataLogic({ context: 'in-app', exportToken: props.exportToken }),
+            [
+                'setHref',
+                'setHrefMatchType',
+                'setHeatmapFilters',
+                'setHeatmapFixedPositionMode',
+                'setHeatmapColorPalette',
+                'setCommonFilters',
+            ],
+        ],
     })),
+    actions({
+        setIsLoading: (isLoading: boolean) => ({ isLoading }),
+        setScreenshotUrl: (screenshotUrl: string) => ({ screenshotUrl }),
+        fetchScreenshotUrl: true,
+    }),
+    reducers({
+        isLoading: [false, { setIsLoading: (_, { isLoading }) => isLoading }],
+        screenshotUrl: [null as string | null, { setScreenshotUrl: (_, { screenshotUrl }) => screenshotUrl }],
+    }),
+    selectors({
+        exportedData: [() => [(_, props: ExportedData) => props], (props: ExportedData) => props],
+    }),
+    listeners(({ actions, props }) => ({
+        fetchScreenshotUrl: async () => {
+            if (props.heatmap_context?.heatmap_type !== 'screenshot') {
+                return
+            }
+            actions.setIsLoading(true)
+            try {
+                const response = await fetch(props.heatmap_url ?? '', {
+                    headers: { Authorization: `Bearer ${props.exportToken}` },
+                })
+                if (response.ok) {
+                    const blob = await response.blob()
+                    const objectUrl = URL.createObjectURL(blob)
+                    actions.setScreenshotUrl(objectUrl)
+                } else {
+                    console.error('Failed to fetch screenshot:', response.status)
+                }
+            } catch (error) {
+                console.error('Failed to fetch screenshot:', error)
+            } finally {
+                actions.setIsLoading(false)
+            }
+        },
+    })),
+    afterMount(({ actions, props }) => {
+        if (props.type === ExportType.Heatmap && props.heatmap_url) {
+            if (props.heatmap_context?.heatmap_type === 'screenshot') {
+                actions.fetchScreenshotUrl()
+            } else {
+                actions.setIsLoading(false)
+            }
+            if (props.heatmap_context?.heatmap_data_url) {
+                actions.setHref(props.heatmap_context?.heatmap_data_url)
+                if (isUrlPattern(props.heatmap_context?.heatmap_data_url)) {
+                    actions.setHrefMatchType('pattern')
+                } else {
+                    actions.setHrefMatchType('exact')
+                }
```

Comment:
nitpick we could listen to setHref and setHrefMatchType in a listener instead of manually here